src/nat/llm/huggingface_llm.py

```
@@ -0,0 +1,286 @@
+                cached["torch"].cuda.empty_cache()
+            
+            # Remove from cache
+            del _model_cache[model_name]
+            
+            logger.debug("Model cleaned up: %s", model_name)
+    except Exception as e:
+        logger.exception("Error cleaning up HuggingFace model '%s': %s", model_name, e)
+
+
+@register_llm_provider(config_type=HuggingFaceConfig)
+async def huggingface_provider(config: HuggingFaceConfig, builder: Builder):
```

Ruff (0.14.8)

Unused function argument: `builder`

(ARG001)